app/models/dscc_number.rb

```diff
@@ -0,0 +1,29 @@
+class DsccNumber < ActiveRecord::Base
+
+  def self.current_monthly_number
+    where(year_and_month: current_year_and_month).order('number desc').limit(1).pluck(:number).first || 0
```

Comment:
Isn't this a massive race condition? If two requests are generated at the same time then they would end up with the same `current_monthly_number` wouldn't they?